src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -4406,17 +4406,17 @@ int32_t NaClSysWaitpid(struct NaClAppThread *natp,
   if (stat_loc_ptr) {
     *stat_loc_ptr = 0;
   }
-  for (int i = 0; i < nap->num_children; i++)
-    pid_max = pid_max < nap->children_ids[i] ? nap->children_ids[i] : pid_max;
-  if (!nap->num_children || cage_id > pid_max) {
+
+  if (!nap->num_children || pid > pid_max) {
     ret = -NACL_ABI_ECHILD;
     goto out;
   }
 
   /*
-   * explicit child pid given
+   * WAITPID: explicit child pid given
    */
-  if (cage_id > 0 && cage_id <= pid_max) {
+  if (pid > 0 && pid <= pid_max) {
+    int cage_id = pid;
```

Comment:
What is the purpose of creating a new variable, cage_id here? Can't you just change the below calls to use pid as an arg?

Reply:
So I did it this way to try to keep some consistency with the WAIT_ANY protocol that also uses cage_id. In my head `pid` is a term we're using in the POSIX context where `cage_id` is what we're using to deal with our internal structures. Maybe its not useful, or I should note this.